src/engine/QueryPlanner.cpp

```diff
@@ -875,6 +876,23 @@ vector<QueryPlanner::SubtreePlan> QueryPlanner::seedWithScansAndText(
           "necessary also rebuild the index.");
     }
 
+    const string keyword = "MaxDistanceInMeter:";
+    const string input = node.triple_._p._iri;
+    if (input.find(keyword) != std::string::npos ) {
+      int maxDist = 0;
+      try {
+      maxDist = std::stoi(input.substr(keyword.size()+1,
+          input.size() - keyword.size()-2));  // -2 to compensate for < and >
+      } catch(const std::exception& e) {
+        LOG(INFO) << "converstion of maxDistance was not possible "
+                  << "assuming infinite maxDistance" << std::endl;
+        LOG(INFO) << "exception: " << e.what() << std::endl;
+      }
```

Comment:
1. `0` shouldn't mean `infinity`, its quite the opposite.
2. You should probably just throw an exception if the parsing fails.
3. The parsing should be a separate function e.g. `std::optional<size_t> parseMaxDistPredicate()` that just returns the distance or nothing. This also increases the testability of that code.
4. The current implementation is wrong, if you just check for contains, then the input `<somethingBlablaMaxDistInMeters:234>` will give you unexpected results. Best use a regex engine (e.g. `ctre`, we have several usages across the codebase), And perform the matching via a regex.)